src/Scrambler.js

```diff
@@ -55,69 +67,79 @@ export default class Scrambler {
 
   _encode() {
     if (this.frameIndex === 0) {
-      const finished = this.encodingCounters.reduce((acc, crr) => acc + crr, 0) === 0;
+      const finished =
+        this.encodingCounters.reduce((acc, crr) => acc + crr, 0) === 0;
+
       if (finished) {
         this.frameId = requestAnimationFrame(() => this._fill());
         return;
       }
 
-      for (let i = 0; i < this.encodingCounters.length; i += 1) {
+      for (let i = 0; i < this.encodingCounters.length; i++) {
         if (this.encodingCounters[i] === 0) {
           const temp = this.scrambledText.split('');
           temp[i] = this._randomText(1);
           this.scrambledText = temp.join('');
           continue;
         }
-        this.encodingCounters[i] -= 1;
+
+        this.encodingCounters[i]--;
         this.onScramble(this.scrambledText);
       }
     }
 
-    this.frameIndex = (this.frameIndex + 1) % 3;
+    this.frameIndex = (this.frameIndex + 1) % this.delay;
```

Comment:
Originally, the values `2`, `3`, `4` were used as delay values to give different speeds to each animation stage (`encode`, `fill`, `decode`). Now that these have been unified under the `delay` value, what do you think about using `delay - 1`, `delay`, and `delay + 1` to maintain that timing effect?